commit 6efb62b8b94405cfd0afecbf2d04985b6f44419b
Author: Jakub Hrozek <jhrozek@redhat.com>
Date:   Tue May 22 17:41:52 2012 +0200

    LDAP nested groups: Do not process callback with _post deep in the nested structure
    
    https://fedorahosted.org/sssd/ticket/1343

diff --git a/src/providers/ldap/sdap_async_groups.c b/src/providers/ldap/sdap_async_groups.c
index 84eb7cb..b587f66 100644
--- a/src/providers/ldap/sdap_async_groups.c
+++ b/src/providers/ldap/sdap_async_groups.c
@@ -2498,14 +2498,13 @@ static errno_t sdap_nested_group_lookup_user(struct tevent_req *req,
                 ret = sdap_nested_group_process_step(req);
             }
 
-            if (ret == EOK) {
-                /* EOK means it's complete */
-                tevent_req_done(req);
-                tevent_req_post(req, state->ev);
-            } else if (ret != EAGAIN) {
+            if (ret != EOK && ret != EAGAIN) {
+                DEBUG(SSSDBG_OP_FAILURE, ("Nested group processing failed\n"));
                 return ret;
+            } else if (ret == EOK) {
+                DEBUG(SSSDBG_TRACE_FUNC, ("All done.\n"));
+                tevent_req_done(req);
             }
-
             return EOK;
         }
         /*
@@ -2583,14 +2582,13 @@ static errno_t sdap_nested_group_lookup_group(struct tevent_req *req)
             ret = sdap_nested_group_process_step(req);
         }
 
-        if (ret == EOK) {
-            /* EOK means it's complete */
-            tevent_req_done(req);
-            tevent_req_post(req, state->ev);
-        } else if (ret != EAGAIN) {
+        if (ret != EOK && ret != EAGAIN) {
+            DEBUG(SSSDBG_OP_FAILURE, ("Nested group processing failed\n"));
             return ret;
+        } else if (ret == EOK) {
+            DEBUG(SSSDBG_TRACE_FUNC, ("All done.\n"));
+            tevent_req_done(req);
         }
-
         return EOK;
     }
 
